src/topology/metric_space/pi_Lp.lean

```diff
@@ -229,6 +229,11 @@ lemma norm_eq {p : ℝ} {hp : 1 ≤ p} {α : ι → Type*}
   [∀i, normed_group (α i)] (f : pi_Lp p hp α) :
   ∥f∥ = (∑ (i : ι), ∥f i∥ ^ p) ^ (1/p) := rfl
 
+lemma norm_eq_of_nat {p : ℝ} {hp : 1 ≤ p} {α : ι → Type*}
+  [∀i, normed_group (α i)] (n : ℕ) (h : p = n) (f : pi_Lp p hp α) :
```

Comment:
Do you need the `p` argument at all? Why not just pass `n` directly into `pi_Lp`?

Reply:
Usually, terms of `pi_Lp ...` are provided like `(x : pi_Lp 2 one_le_two f)` as in `pi_Lp.norm_eq_of_L2`. There, the `2` is a real. If one was to use `n` directly and have `hn : 1 ≤ n`, `(f : pi_Lp n (nat.one_le_cast.mpr hn) α)` then the `f` the lemma would be expecting would look like `(f : pi_Lp ↑2 _ α)` which is no longer defeq and can't be used to rewrite.

Reply:
@eric-wieser I'm guessing this is to avoid DTT problems later on?  That is, `pi_Lp p` being different from `pi_Lp (↑n)`.

Reply:
Thanks for explaining!